marketdata/stream/client_test.go

```diff
@@ -600,6 +601,160 @@ func TestSubscriptionAcrossConnectionIssues(t *testing.T) {
 	require.ElementsMatch(t, []string{"PACA"}, c.sub.trades)
 }
 
+func TestSubscriptionTwiceAcrossConnectionIssues(t *testing.T) {
+	mockTimeAfterCh := make(chan time.Time)
+	timeAfter = func(d time.Duration) <-chan time.Time {
+		return mockTimeAfterCh
+	}
+	defer func() {
+		timeAfter = time.After
+	}()
+
+	conn1 := newMockConn()
+	writeInitialFlowMessagesToConn(t, conn1, subscriptions{})
+
+	connected := make(chan struct{})
+	connectCallback := func() {
+		t.Log("connected")
+		connected <- struct{}{}
+	}
+
+	disconnected := make(chan struct{})
+	disconnectCallback := func() {
+		t.Log("disconnected")
+		disconnected <- struct{}{}
+	}
+
+	key := "testkey"
+	secret := "testsecret"
+	c := NewStocksClient(marketdata.IEX,
+		WithCredentials(key, secret),
+		withConnCreator(func(ctx context.Context, u url.URL) (conn, error) {
+			return conn1, nil
+		}),
+		WithReconnectSettings(0, 150*time.Millisecond),
+		WithConnectCallback(connectCallback),
+		WithDisconnectCallback(disconnectCallback),
+	)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// connect
+	err := c.Connect(ctx)
+	require.NoError(t, err)
+	// wait connect callback
+	<-connected
+	checkInitialMessagesSentByClient(t, conn1, key, secret, subscriptions{})
+
+	// subscribing to something
+	trades1 := []string{"AL", "PACA"}
+	subRes := make(chan error)
+	subFunc := func() {
+		subRes <- c.SubscribeToTrades(func(trade Trade) {}, "AL", "PACA")
+	}
+	go subFunc()
+	sub := expectWrite(t, conn1)
+	require.Equal(t, "subscribe", sub["action"])
+	require.ElementsMatch(t, trades1, sub["trades"])
+	// server accepts subscription
+	conn1.readCh <- serializeToMsgpack(t, []subWithT{
+		{
+			Type:   "subscription",
+			Trades: trades1,
+		},
+	})
+	err = <-subRes
+	require.NoError(t, err)
+
+	// shutting down the first connection
+	c.connCreator = func(ctx context.Context, u url.URL) (conn, error) {
+		return nil, fmt.Errorf("connection failed")
+	}
+	conn1.close()
+	// wait disconnect callback
+	<-disconnected
+
+	// request subscribe will be timed out during disconnection
+	go subFunc()
+
+	mockTimeAfterCh <- time.Now()
+	err = <-subRes
+	assert.Error(t, err)
+	assert.True(t, errors.Is(err, ErrSubscriptionChangeTimeout), "actual: %s", err)
+
+	// after a timeout we should be able to get timed out again
+	go subFunc()
+
+	mockTimeAfterCh <- time.Now()
+	err = <-subRes
+	assert.Error(t, err)
+	assert.True(t, errors.Is(err, ErrSubscriptionChangeTimeout), "actual: %s", err)
+
+	// establish 2nd connection
+	conn2 := newMockConn()
+	writeInitialFlowMessagesToConn(t, conn2, subscriptions{trades: trades1})
+	// writeInitialFlowMessagesToConn(t, conn2, subscriptions{})
```

Comment:
[minor] Please remove the commented code.

Reply:
Thank you! missed it. Removed https://github.com/alpacahq/alpaca-trade-api-go/pull/274/commits/80878d871ff2085a328485185fefd614277af525